cmd/skaffold/app/cmd/flags.go

```diff
@@ -17,243 +17,164 @@ limitations under the License.
+		},
 	},
 	{
-		Name:          "force",
-		Usage:         "Recreate kubernetes resources if necessary for deployment (warning: might cause downtime!)",
-		Value:         &opts.ForceDev,
-		DefValue:      true,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "run", "debug"},
+		DefinedOn: []string{"dev", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.ForceDev, "force", true, "Recreate kubernetes resources if necessary for deployment (warning: might cause downtime!)")
+		},
 	},
 	{
-		Name:          "skip-tests",
-		Usage:         "Whether to skip the tests after building",
-		Value:         &opts.SkipTests,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "run", "debug", "build"},
+		DefinedOn: []string{"dev", "run", "debug", "build"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.SkipTests, "skip-tests", false, "Whether to skip the tests after building")
+		},
 	},
 	{
-		Name:          "cleanup",
-		Usage:         "Delete deployments after dev or debug mode is interrupted",
-		Value:         &opts.Cleanup,
-		DefValue:      true,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "run", "debug"},
+		DefinedOn: []string{"dev", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.Cleanup, "cleanup", true, "Delete deployments after dev or debug mode is interrupted")
+		},
 	},
 	{
-		Name:          "no-prune",
-		Usage:         "Skip removing images and containers built by Skaffold",
-		Value:         &opts.NoPrune,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "run", "debug"},
+		DefinedOn: []string{"dev", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.NoPrune, "no-prune", false, "Skip removing images and containers built by Skaffold")
+		},
 	},
 	{
-		Name:          "no-prune-children",
-		Usage:         "Skip removing layers reused by Skaffold",
-		Value:         &opts.NoPruneChildren,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "run", "debug"},
+		DefinedOn: []string{"dev", "run", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.NoPruneChildren, "no-prune-children", false, "Skip removing layers reused by Skaffold")
+		},
 	},
 	{
-		Name:          "port-forward",
-		Usage:         "Port-forward exposed container ports within pods",
-		Value:         &opts.PortForward,
-		DefValue:      false,
-		FlagAddMethod: "BoolVar",
-		DefinedOn:     []string{"dev", "debug"},
+		DefinedOn: []string{"dev", "debug"},
+		Setup: func(f *pflag.FlagSet) {
+			f.BoolVar(&opts.PortForward, "port-forward", false, "Port-forward exposed container ports within pods")
+		},
 	},
 }
 
-var commandFlags []*pflag.Flag
-
-// SetUpFlags creates pflag.Flag for all registered flags
-func SetUpFlags() {
```

Comment:
I was personally okay with having the reflection based logic to have the nice struct.
It is admittedly a bit of a hack in the sense that it uses reflect and that it bets on the shared structure of the  flagset API but results in a named structure that I find more readable/usable than the single line version of the flags and that is the key for me. I guess this is one of those "opinion" based decisions. Do you feel strongly about this direction?

Reply:
Yeah, I felt bad of kinda rolling back @tejal29´s effort. I personally like the fact that most flags are now defined globally. I didn’t really like the super verbose structure that uses reflection when every flag is a one liner. But as you say, it’s a matter taste. I’m happy to remove that part of the PR if you prefer

Reply:
@tejal29 wdyt? I'm okay either way.

Reply:
I think the motivation for making it more verbose was so that the flags are easy to read.
Right now,  one needs to dig deeper in the method signature to identify what is &opts.PortForward, "false" or "Port-forward exposed container ports within pods"

```
f.BoolVar(&opts.PortForward, "port-forward", false, "Port-forward exposed container ports within pods")
```

I had other iterations on my branch where i used BoolVar and related methods.

Reply:
I agree - for me too, it looks like that the space-saving reduces readability. Let's revert to structs, if you don't mind, @dgageot?

Reply:
I'll do that!